keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+          resources:
+            requests:
+              storage: 250Gi
+ ---
+ apiVersion: v1
+ kind: Service
+ metadata:
+   name: wordpress-webserver-svc
+   labels:
+     app: wordpress
+     component: wordpress-webserver-svc
+ spec:
+   ports:
+     - port: 80
+   selector:
+     app: wordpress
+     component: wordpress-webserver
+   type: LoadBalancer
+ ---
+ apiVersion: v1
+ kind: Service
+ metadata:
+   name: wordpress-webserver-hsvc
+   labels:
+     app: wordpress
+     component: wordpress-webserver-hsvc
+ spec:
+   ports:
+     - port: 3306
+   selector:
+     app: wordpress
+     component: wordpress-webserver
+   clusterIP: None
+ ---
+ apiVersion: apps/v1
+ kind: StatefulSet
+ metadata:
+   name: wordpress
+   labels:
+     app: wordpress
+     component: wordpress-webserver
+ spec:
+   replicas: 1
+   service: wordpress-webserver-hsvc
+   selector:
+     matchLabels:
+       app: wordpress
+       component: wordpress-webserver
+   template:
+     metadata:
+       labels:
+         app: wordpress
+         tier: frontend
+     spec:
+       containers:
+       - image: wordpress:4.8-apache
+         name: wordpress
+         env:
+         - name: WORDPRESS_DB_HOST
+           value: wordpress-mysql
+         - name: WORDPRESS_DB_PASSWORD
+           valueFrom:
+             secretKeyRef:
+               name: mysql-pass
+               key: password
+         ports:
+         - containerPort: 80
+           name: wordpress
+         volumeMounts:
+         - name: wordpress-persistent-storage
+           mountPath: /var/www/html
+   volumeClaimTemplates:
+        - metadata:
+            name: mysql-persistent-storage
+          spec:
+            accessModes: [ "ReadWriteOnce" ]
+            resources:
+              requests:
+                storage: 250Gi
+ ```
+ 
+From an infrastructure management perspective, the manifest above encapsulates everything that is necessary to create 
+the workloads and services that compose our WordPress CMS application. However, the fact that these workloads comprise 
+an application is lost in translation during creation. Once these manifests are applied, there is no default way 
+for users to reconstruct the fact that the whole is more than the sum of its parts.
+ 
+When UIs, either command line or graphical, seek to interact with the application, they can only interact with its 
+components, and there is no standard way to indicate that those components comprise an application. As previously 
+mentioned, Many tools in the ecosystem attempt to aggregate the components of an application using labels and/or 
+annotations, but, as each tool uses a different scheme, an application created by one tool is generally not recognizable 
+by another, and UI designers have no standard way to recognize and display an application created by any tool.
+
+Moreover, their is no way to determine if the application is healthy. Even if a UI is able to determine the components 
```

Comment:
I believe in general there is a need to summarize the status of an "application" instead of its individual components at low level. It could be flexible enough so users can define how they want to summarize it, but I'd like to see how the proposal addresses it concretely.